surch/surch.py

```diff
@@ -1,165 +1,330 @@
-        for repo_data in data:
-            github_repo_urls.append(repo_data['clone_url'])
-        return github_repo_urls
-
-    def clone_repo(self, github_repo_urls, path):
-        ''' clone or pull git repos'''
+        conf_vars.setdefault('verbose', verbose)
+        conf_vars.setdefault('quiet_git', quiet_git)
+        return cls(**conf_vars)
+
+    def _create_local_directory_if_not_exsist(self, path):
+        if not os.path.isdir(path):
+            os.makedirs(path)
+
+    def get_github_repo_list(self, repo_type='public', repos_per_page=100):
+        """ This method get from git hub the git url list for clonnig
+
+        :param repo_type: repository type (all, private, public, fork) default:'public'
+        :type repo_type: basestring
+
+        :param repos_per_page: this for getting the MAX information in 1 page default:100
+        :type repos_per_page: int
+        :return:
+        """
+        self.all_data = []
+        all_data = requests.get(API_URL_ORGANIZATION_DETAILS.format(self.organization),
+                                auth=(self.git_user, self.git_password))
+
+        repos_number = all_data.json()['{0}_repos'.format(repo_type)]
+        last_page_number = repos_number / repos_per_page
+        if (repos_number % repos_per_page) > 0:
+            # Adding 2 because 1 for the extra repos that mean more page, and 1 for the next for loop.
+            last_page_number += 2
+
+            for page_num in range(1, last_page_number):
+                all_data = requests.get(API_URL_REPO_DETAILS.format(self.organization, repo_type,
+                                                                    repos_per_page, page_num),
+                                        auth=(self.git_user, self.git_password))
+                for repo in all_data.json():
+                    self.all_data.append(repo)
+
+    def clone_repo(self, local_path, url_type='clone_url'):
+        """ This method run clone or pull for the repo list
+
+        :param local_path: this path contain the repos clone
+        :type local_path: basestring
+        :param url_type: url type (git_url, ssh_url, clone_url, svn_url) default:'clone_url'
+        :type url_type: basestring
+        :return: cloned repo
+        """
+        start = time.time()
+        for repo in self.all_data:
+            if repo['name'] not in self.ignore_repo:
+                full_path = os.path.join(local_path, repo['name'])
+                self._clone_or_pull(full_path, repo['name'], repo[url_type])
+        lgr.info('git clone\pull time: ' + print_performance(start, time.time()) + ' seconds')
+
+    @retrying.retry(stop_max_attempt_number=3)
+    def _clone_or_pull(self, full_path, repository_name, url):
+        """ This method check if the repo exsist in the path and run clone or pull"""
+
+        if os.path.isdir(full_path):
+            try:
+                lgr.info('Pull {0} repo, from {1} org/user.'.format(repository_name, self.organization))
```

Comment:
remove from... and add logging of the organization name in the beginning of the process